src/plugins/plot/chart/MCTChartSeriesElement.js

```diff
@@ -72,21 +73,22 @@ export default class MCTChartSeriesElement {
         }
     }
 
-    removePoint(point, index, count) {
-        // by default, do nothing.
-    }
+    /** @abstract */
+    removePoint(index) {}
+
+    /** @abstract */
+    addPoint(point, index) {}
 
     remove(point, index, series) {
         const vertexCount = this.vertexCountForPointAtIndex(index);
         const removalPoint = this.startIndexForPointAtIndex(index);
 
         this.removeSegments(removalPoint, vertexCount);
 
-        this.removePoint(
-            this.makePoint(point, series),
-            removalPoint,
-            vertexCount
-        );
+        // TODO useless makePoint call?
+        this.makePoint(point, series);
+        this.removePoint(removalPoint);
+
```

Comment:
What's the point of this `makePoint` call? I left it in place just in case there's a side-effect I don't know about (is there?), but `this.removePoint()` is not using it.

Seems like copy/paste left overs from a long time ago. What's the point of making a point just to remove a another point?

Reply:
@shefalijoshi can you take a look at this?